### ISSUE TITLE
error in renaming file names  

### ISSUE THREAD

**Issue by shSina** [flyingrub/scdl#56]
cannot download this file as an example :
https://soundcloud.com/radiobooom/yaad
```shell
Found a track
Downloading یـــــــــاد
Downloading the orginal file.
100% [..........................................................................] 6410120 / 6410120Traceback (most recent call last):
  File "/usr/lib/python3.4/shutil.py", line 522, in move
    os.rename(src, real_dst)
FileNotFoundError: [Errno 2] No such file or directory: '/home/sina/scdl/download.qsb5lwo6.tmp' -> "Ì@@@@@@@@@'/.mp3"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/scdl", line 9, in <module>
    load_entry_point('scdl==v1.3.2', 'console_scripts', 'scdl')()
  File "/usr/local/lib/python3.4/dist-packages/scdl/scdl.py", line 115, in main
    parse_url(arguments["-l"])
  File "/usr/local/lib/python3.4/dist-packages/scdl/scdl.py", line 179, in parse_url
    download_track(item)
  File "/usr/local/lib/python3.4/dist-packages/scdl/scdl.py", line 398, in download_track
    wget.download(url, filename)
  File "/usr/local/lib/python3.4/dist-packages/wget.py", line 320, in download
    shutil.move(tmpfile, filename)
  File "/usr/lib/python3.4/shutil.py", line 534, in move
    copy2(src, real_dst)
  File "/usr/lib/python3.4/shutil.py", line 244, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib/python3.4/shutil.py", line 108, in copyfile
    with open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: "Ì@@@@@@@@@'/.mp3"
```

**Comment by flyingrub** [comment]
actually this is related to wget not scdl ...

**Comment by lucasrangit** [comment]
@shSina I attached a file with non-roman characters to the ticket on BitBucket. Can you you verify that you can download this file and rename it from a terminal on your computer either using `mv` or Python `on.rename()`?

**Comment by shSina** [comment]
ubuntu terminal does not support showing unicode(rtl) characters , but it can rename the file name with correct copied characters 
<img width="507" alt="screen shot 2015-09-02 at 10 36 13 pm" src="https://cloud.githubusercontent.com/assets/478063/9639808/1e6f356e-51c3-11e5-98a3-db8706ebc857.png">
mac os terminal support showing unicode(rtl) characters correctly as well as successful rename.
<img width="402" alt="screen shot 2015-09-02 at 10 42 09 pm" src="https://cloud.githubusercontent.com/assets/478063/9639960/10f0ea9e-51c4-11e5-82ec-0a5d9a7e86d6.png">

**Comment by flyingrub** [comment]
@lucasrangit i tested and could rename the file with both mv and os.rename but there is still an error with the script.

**Comment by techtonik** [comment]
I am looking into that right now.

Can you `print(type(filename))` before `shutil.move(tmpfile, filename)` in wget.py? If it is bytes, then it could be fixed. If not, then the issue may be worse.

**Comment by techtonik** [comment]
[x] Windows + Python 2
doesn't fail with error, but created file is named `ÛŒÙ€Ù€Ù€Ù€Ù€Ù€Ù€Ù€Ù€Ø§Ø¯.zip`
[x] Windows + Python 3
fails with this
```
Traceback (most recent call last):
  File "file.down.py", line 3, in <module>
    wget.download('http://localhost:8000/test/\u06cc\u0640\u0640\u0640\u0640\u0640\u0640\u0640\u0640\u0640\u0627\u062f.zip')
  File "C:\wget\wget\wget.py", line 308, in download
    (tmpfile, headers) = urllib.urlretrieve(url, tmpfile, callback)
  File "C:\Python34\lib\urllib\request.py", line 178, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
  File "C:\Python34\lib\urllib\request.py", line 153, in urlopen
    return opener.open(url, data, timeout)
  File "C:\Python34\lib\urllib\request.py", line 455, in open
    response = self._open(req, data)
  File "C:\Python34\lib\urllib\request.py", line 473, in _open
    '_open', req)
  File "C:\Python34\lib\urllib\request.py", line 433, in _call_chain
    result = func(*args)
  File "C:\Python34\lib\urllib\request.py", line 1215, in http_open
    return self.do_open(http.client.HTTPConnection, req)
  File "C:\Python34\lib\urllib\request.py", line 1189, in do_open
    h.request(req.get_method(), req.selector, req.data, headers)
  File "C:\Python34\lib\http\client.py", line 1090, in request
    self._send_request(method, url, body, headers)
  File "C:\Python34\lib\http\client.py", line 1118, in _send_request
    self.putrequest(method, url, **skips)
  File "C:\Python34\lib\http\client.py", line 975, in putrequest
    self._output(request.encode('ascii'))
UnicodeEncodeError: 'ascii' codec can't encode characters in position 10-21: ordinal not in range(128)
```
Test file:
```
import wget

wget.download('http://localhost:8000/test/یـــــــــاد.zip')
```

**Comment by techtonik** [comment]
Committed the fix for Python 2. Need to spend some more cycles to find the solution the Python 3.

**Comment by techtonik** [comment]
A known problem of 3rd Python - https://stackoverflow.com/questions/4389572/how-to-fetch-a-non-ascii-url-with-python-urlopen Figuring out the workaround.

**Comment by flyingrub** [comment]
This soft doesn't support python2 and the url is fetch correctly. As it print the right name in the terminal.

**Comment by techtonik** [comment]
@flyingrub do you encode unicode URL to 'utf-8' somehow before passing to wget? Because I stuck on this part, far from saving file to disk.

**Comment by flyingrub** [comment]
are you on windows ?

**Comment by techtonik** [comment]
@flyingrub yes.

**Comment by techtonik** [comment]
Just released 3.1 which should fix your issue.

**Comment by flyingrub** [comment]
@shSina if the error still occur let me know.
